src/components/schedule-page/table/schedule/schedule-parts/base-cell/cell-details-content.component.tsx

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import React, { useContext } from "react";
+import { TranslationHelper } from "../../../../../../helpers/translations.helper";
+import { shifts } from "../../../../../../common-models/shift-info.model";
+import { MdClose } from "react-icons/md";
+import { ScheduleLogicContext } from "../../use-schedule-state";
+import { Sections } from "../../../../../../logic/providers/schedule-provider.model";
+import { ShiftsInfoLogic } from "../../../../../../logic/schedule-logic/shifts-info.logic";
+
+export interface CellDetailsOptions {
+  index: number;
+  day: number;
+  month: number;
+  year: number;
+  rowIndex: number;
+  shift: string;
+  sectionKey: string;
+  close: () => void;
+}
+
+export function CellDetails(props: CellDetailsOptions): JSX.Element {
+  const { index, day, month, year, shift, close, sectionKey, rowIndex } = props;
+  let displayedYear = year;
+  let monthName = `${TranslationHelper.polishMonthsGenetivus[month]}`;
+  if (index < day) {
+    if (month > 0) {
+      monthName = `${TranslationHelper.polishMonthsGenetivus[month - 1]}`;
+    } else {
+      monthName = `${TranslationHelper.polishMonthsGenetivus[month + 11]}`;
+      displayedYear = year - 1;
+    }
```

Comment:
Tutaj ta cała logika jest chyba prostsza, ale trzeba przetestować
```const monthName = `${TranslationHelper.polishMonthsGenetivus[((index < day ? 11 : 0) + month) % 12]}```;
to jest w sumie fajny kawałek kodu, żeby zrobić z tego funkcję małą i napisać test jednostkowy, bo logika jest co najmniej dziwna czytająć kod, nie wiedząć co to index i day

Reply:
Chyba nie da się tego zrobić tak prosto, bo mamy trzy przypadki i w zależności od nich zmieniamy i miesiąc i rok. Ale dzięki temu, że napisałeś zauważyłam, że wcześniej miałam tu błąd [psuło się w lutym].

![month and year](https://user-images.githubusercontent.com/48785655/105986020-9a9bd280-609c-11eb-9646-5e4de589523e.png)

Reply:
Okej, no trudno, ale jak się nie da tego uprościć to bym zrobił z tego funkcję jakąś przykładowo, żeby ją uruchamiać w taki sposób:
` const [displayedYear, monthName] = prepareYearAndMonth(index, day, month)`
oczywiście nazwę możesz zmienić na jakąś inną/lepszą

Reply:
poprawiłam :)